sdk/src/test/java/com/qonversion/android/sdk/internal/di/network/NetworkAssemblyTest.kt

```diff
@@ -158,47 +161,48 @@ internal class NetworkAssemblyTest {
                     policy
                 )
                 // when
-                val result = networkAssembly.provideApiInteractor(policy)
+                val result = networkAssembly.apiInteractor(policy)
 
                 // then
+                assertThat(result).isInstanceOf(ApiInteractorImpl::class.java)
                 assertThat(result).isEqualToComparingFieldByField(expectedResult)
             }
         }
 
         @Test
         fun `get exponential api interactor`() {
             // given
-
             val mockApiInteractor = mockk<ApiInteractor>()
+            val policySlot = slot<RetryPolicy>()
 
             every {
-                networkAssembly.exponentialApiInteractor()
+                networkAssembly.apiInteractor(capture(policySlot))
             } returns mockApiInteractor
 
             // when
             val result = networkAssembly.exponentialApiInteractor()
 
             // then
-
             assertThat(result).isSameAs(mockApiInteractor)
+            assertThat(policySlot.captured).isInstanceOf(RetryPolicy.Exponential::class.java)
         }
 
         @Test
         fun `get infinite exponential api interactor`() {
             // given
-
             val mockApiInteractor = mockk<ApiInteractor>()
+            val policySlot = slot<RetryPolicy>()
 
             every {
-                networkAssembly.infiniteExponentialApiInteractor()
+                networkAssembly.apiInteractor(capture(policySlot))
             } returns mockApiInteractor
 
             // when
             val result = networkAssembly.infiniteExponentialApiInteractor()
 
             // then
-
             assertThat(result).isSameAs(mockApiInteractor)
+            assertThat(policySlot.captured).isInstanceOf(RetryPolicy.InfiniteExponential::class.java)
         }
```

Comment:
I think we should use `spyk` in these tests and mock the call of `apiInteractor` with the correct retry policy and then just verify that that call happened. What do you think?